### PR TITLE
Fix 0104 date parsing crash

### PR DESCRIPTION
**Pull request by daniel-riehm**
Change behavior when a ST0104 date fails to parse from crashing via uncaught exception to printing an error message and continuing on.

### CODE REVIEW

**Comment by daniel-riehm** [comment]
@hdefazio

File: arrows/klv/klv_convert_vital.cxx [hunk 1]
```diff
@@ -241,19 +241,33 @@ klv_0104_to_vital_metadata( klv_timeline const& klv_data, uint64_t timestamp,
...
+    {
+      auto const value = event_start_datetime.get< std::string >();
+      vital_data.add< kv::VITAL_META_EVENT_START_TIMESTAMP >(
+          kv::std_0104_datetime_to_unix_timestamp( value ) );
+    }
+    catch( kv::metadata_exception const& e )
+    {
+      LOG_ERROR( kv::get_logger( "klv" ), e.what() );
+    }
   }
```
Comment (hdefazio) [inline arrows/klv/klv_convert_vital.cxx:271]: This seems repetitive with the if statement above - possibly make into a function?

File: arrows/klv/klv_convert_vital.cxx [hunk 2]
```diff
@@ -236,25 +261,13 @@ klv_0104_to_vital_metadata( klv_timeline const& klv_data, uint64_t timestamp,
...
-  // Parse the datetime string into a UNIX microsecond timestamp
-  auto const event_start_datetime =
-    klv_data.at( standard, KLV_0104_START_DATETIME, timestamp );
-  if( start_datetime.valid() )
-  {
-    auto const value = event_start_datetime.get< std::string >();
-    vital_data.add< kv::VITAL_META_EVENT_START_TIMESTAMP >(
-        kv::std_0104_datetime_to_unix_timestamp( value ) );
-  }
+  // Parse the datetime strings into UNIX microsecond timestamps
```
Comment (hdefazio) [inline arrows/klv/klv_convert_vital.cxx:264]: This comment might be better as the function description. And maybe rename the function to `klv_0104_parse_datetime_to_unix` so the comment isn't needed here?

Comment (daniel-riehm) [reply arrows/klv/klv_convert_vital.cxx:264]: I use these comments to break up this large function into "sections" to improve readability, sort of like the `//---` function separators. So I would want to keep a comment here, even if not strictly necessary for its own sake. But I can change the function name if you think it improves comprehension at function definition.

Comment (hdefazio) [reply arrows/klv/klv_convert_vital.cxx:264]: yep, that's fine. This looks good to me now.

**Pull request merged**